framework/include/utils/RankTwoTensorImplementation.h

```diff
@@ -1125,26 +1141,19 @@ template <typename T>
 RankTwoTensorTempl<T>
 RankTwoTensorTempl<T>::genRandomTensor(T scale, T offset)
 {
-  RankTwoTensorTempl<T> tensor;
-
+  RankTwoTensorTempl<T> result;
   for (auto i : make_range(N))
     for (auto j : make_range(N))
-      tensor(i, j) = (MooseRandom::rand() + offset) * scale;
-
-  return tensor;
+      result(i, j) = (MooseRandom::rand() + offset) * scale;
```

Comment:
```suggestion
      result(i, j) = MooseRandom::randNormal(offset, scale);
```
Wasn't the original implementation wrong?! Should've been `MooseRandom::rand() * scale + offset`.

Reply:
I really don't want to change the existing behavior in this PR :-/

Reply:
Fair enough. I'll make an issue for this. I will fix this later. I hope no internal app was using this function.

Reply:
Noted in #21269